presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -107,6 +125,65 @@ Property Name                                      Description
+                                                   Example: ``hive/hive-server-host@EXAMPLE.COM`` or
+                                                   ``hive/_HOST@EXAMPLE.COM``.
+
+``hive.metastore.client.principal``                Hive metastore client principal.
+                                                   The ``_HOST`` placeholder is allowed here and it is
+                                                   substituted with the actual Presto server host. Use
+                                                   ``_HOST`` placeholder for the principal per server
+                                                   configurations.
+                                                   Example: ``presto/presto-server-node@EXAMPLE.COM`` or
+                                                   ``presto/_HOST@EXAMPLE.COM``.
+
+                                                   .. note::
+
+                                                        Deletion of the internal table data is the
+                                                        responsibility of Hive metastore. When metastore is
+                                                        configured to use Kerberos authentication, all the
+                                                        HDFS operations performed by metastore are
+                                                        impersonated. Make sure that provided user has
+                                                        enough permission to remove the folders within
+                                                        the ``hive/warehouse`` directory. Otherwise data
+                                                        removal is silently skipped by metastore and
+                                                        only the metadata records are removed.
+
+``hive.metastore.client.keytab``                   Hive metastore client keytab location. Must be accessible
+                                                   for the user running Presto and must contain the
+                                                   credentials for the  ``hive.metastore.client.principal``.
+
+``hive.hdfs.authentication.type``                  HDFS authentication type.                                    ``SIMPLE``
+                                                   Possible values are ``SIMPLE`` or ``KERBEROS``.
+
+``hive.hdfs.impersonation.enabled``                Enable HDFS calls impersonation. With impersonation          ``false``
+                                                   enabled all the calls to HDFS are performed from the
+                                                   user currently logged in to Presto. With impersonation
+                                                   disabled all the calls to HDFS are performed from the
+                                                   user specified in ``hive.hdfs.presto.principal``.
+
```

Comment:
Not your fault, but the whole situation here is super confusing because there are 4 different kinds of "user-like things":

1. Kerberos principals
2. Operating system users
3. The strings that HDFS calls users in the absence of Kerberos.
4. Presto users (--user)

SIMPLE, false: HDFS access as the OS user of the presto process or the value of -DHADOOP_USER_NAME if specified.
SIMPLE, true: HDFS access as --user.
KERBEROS, false: HDFS access as `hive.hdfs.presto.principal`
KERBEROS, true: HDFS access as --user.

We should try to get something simple and reasonably accurate in here, and I'll add a link to the security/hive.rst when I put up the PR for that.

@BrendaNoonan, feel free to weigh in here too.